packages/protocol/test/identity/federatedattestations.ts

```diff
@@ -235,7 +229,87 @@ contract('FederatedAttestations', (accounts: string[]) => {
   })
 
   describe('#registerAttestation', () => {
-    it('should', async () => {})
+    beforeEach(async () => {
+      await accountsInstance.authorizeSigner(signer, signerRole, { from: issuer })
+      await accountsInstance.completeSignerAuthorization(issuer, signerRole, { from: signer })
+    })
+    it('should emit AttestationRegistered for a valid attestation', async () => {
+      const register = await federatedAttestations.registerAttestation(
+        pnIdentifier,
+        issuer,
+        account,
+        issuedOn,
+        signer,
+        sig.v,
+        sig.r,
+        sig.s
+      )
+      assertLogMatches2(register.logs[0], {
+        event: 'AttestationRegistered',
+        args: {
+          identifier: pnIdentifier,
+          issuer,
+          account,
+          issuedOn,
+          signer,
+        },
+      })
+    })
+
+    it('should revert if the same attestation is uploaded again', async () => {
```

Comment:
nit: maybe something like "should revert if a second attestation for the same (issuer, identifier, account) is uploaded", just to be extra explicit?